circuit-lib/circuit-lib.circom/benchmarks/light/transaction_light.ts

```diff
@@ -0,0 +1,169 @@
+const bench = require("micro-bmark");
+const snarkjs = require("snarkjs");
+const buildPoseidonOpt = require("circomlibjs").buildPoseidonOpt;
+
+import { Keypair as SolanaKeypair, PublicKey } from "@solana/web3.js";
+import { BN } from "@coral-xyz/anchor";
+
+import {
+  Account,
+  Utxo,
+  FEE_ASSET,
+  Provider as LightProvider,
+  MINT,
+  Transaction,
+  TransactionParameters,
+  Action,
+  IDL_VERIFIER_PROGRAM_ZERO,
+  IDL_VERIFIER_PROGRAM_TWO,
+} from "@lightprotocol/zk.js";
+
+import { bs58 } from "@coral-xyz/anchor/dist/cjs/utils/bytes";
+import { MerkleTree } from "../../../circuit-lib.js/lib/index.js";
+import { stringifyBigInts } from "maci-crypto";
+
+// This will extract the type of the constructor parameters of Transaction CLass
+type TransactionInputs = ConstructorParameters<typeof Transaction>[0];
+
+function addPrivateKey(transaction: Transaction, account: Account) {
+  transaction.proofInput["inPrivateKey"] = transaction.params.inputUtxos.map(
+    (utxo: Utxo) => {
+      if (utxo.publicKey == account.pubkey) {
+        return account.privkey;
+      }
+    },
+  );
+}
+
+async function generateProofInputs(tree_height: number, app = false) {
+  const lightProvider = await LightProvider.loadMock();
+  const poseidon = await buildPoseidonOpt();
+
+  const seed32 = bs58.encode(new Uint8Array(32).fill(1));
+  const account = new Account({ poseidon: poseidon, seed: seed32 });
+  await account.getEddsaPublicKey();
+
+  const depositAmount = 20_000;
+  const depositFeeAmount = 10_000;
+
+  const deposit_utxo1 = new Utxo({
+    poseidon: poseidon,
+    assets: [FEE_ASSET, MINT],
+    amounts: [new BN(depositFeeAmount), new BN(depositAmount)],
+    publicKey: account.pubkey,
+    assetLookupTable: lightProvider.lookUpTables.assetLookupTable,
+    verifierProgramLookupTable:
+      lightProvider.lookUpTables.verifierProgramLookupTable,
+  });
+
+  const mockPubkey: PublicKey = SolanaKeypair.generate().publicKey;
+
+  let txParams = new TransactionParameters({
+    outputUtxos: [deposit_utxo1],
```

Comment:
I am surprised that the bench results show a much lower increase in proving time than the results of the merkle proof benches. Can you add as many in utxos as the circuit allows to make sure we get a comparable number?

Merkle proofs are just enforced when amounts of input utxos are non zero and I think for empty utxos we just use zeros as merkle proofs. Thus the benchmarks might not be representative.

Reply:
added 2^13 elements => random Field Elements that simulate hashes.